f8a_report/report_helper.py

```diff
@@ -159,16 +189,86 @@ def normalize_deps_list(self, deps):
+            result.update({eco: dict(
+                Counter(unique_stacks_with_recurrence_count.get(eco)) +
+                Counter(collated_user_input.get(eco)))
+            })
+
+        # Store user input collated data back to S3
+        self.s3.store_json_content(content=result, bucket_name=self.s3.report_bucket_name,
+                                   obj_key=collated_user_input_obj_key)
+
+        # Get collated big query data
+        collated_big_query_obj_key = '{depl_prefix}/big-query-data/collated.json'.format(
+            depl_prefix=self.s3.deployment_prefix)
+        collated_big_query_data = self.s3.read_json_object(bucket_name=self.s3.report_bucket_name,
+                                                           obj_key=collated_big_query_obj_key) or {}
+
+        for eco in result.keys() | collated_big_query_data.keys():
+            result.update({
+                eco: dict(Counter(result.get(eco)) +
+                          Counter(collated_big_query_data.get(eco)))
+                    })
+
+        return result
+
+    def store_training_data(self, result):
+        """Store Training Data for each ecosystem in their respective buckets."""
+        model_version = dt.now().strftime('%Y-%m-%d')
+
+        for eco, stacks in result.items():
+            obj_key = '{eco}/{depl_prefix}/{model_version}/data/manifest.json'.format(
+                eco=eco, depl_prefix=self.s3.deployment_prefix, model_version=model_version)
+            package_list_for_eco = []
+            for packages, reccurrence_count in stacks.items():
+                package_list = [x.strip().split(' ')[0] for x in packages.split(',')]
+                package_list_for_eco.append(package_list)
+
+            training_data = {
+                'ecosystem': eco,
+                'package_list': package_list_for_eco
+            }
+
+            # Get the bucket name based on ecosystems to store user-input stacks for retraining
+            if eco == 'maven':
+                bucket_name = self.maven_model_bucket
+            elif eco == 'pypi':
+                bucket_name = self.pypi_model_bucket
+            elif eco == 'go':
+                bucket_name = self.golang_model_bucket
+            elif eco == 'npm':
+                bucket_name = self.npm_model_bucket
+            else:
+                continue
+
+            if bucket_name:
+                logger.info('Storing user-input stacks for ecosystem {eco} at {dir}'.format(
+                    eco=eco, dir=bucket_name + obj_key))
+                self.s3.store_json_content(content=training_data, bucket_name=bucket_name,
+                                           obj_key=obj_key)
+
     def normalize_worker_data(self, start_date, end_date, stack_data, worker, frequency='weekly'):
         """Normalize worker data for reporting."""
         total_stack_requests = {'all': 0, 'npm': 0, 'maven': 0}
+        report_name = dt.strptime(end_date, '%Y-%m-%d').strftime('%Y-%m-%d')
```

Comment:
@miteshvp  Does this mean that even the monthly report names are going to be '%Y-%m-%d'?

Reply:
Missed it. Added now